test/OpenFeature.Contrib.Hooks.Otel.Test/MetricsHookTest.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using OpenFeature.Model;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using Xunit;
+
+namespace OpenFeature.Contrib.Hooks.Otel.Test
+{
+    public class MetricsHookTest
+    {
+        [Fact]
+        public void After_Test()
+        {
+            // Arrange metrics collector
+            var exportedItems = new List<Metric>();
+            Sdk.CreateMeterProviderBuilder()
+                .AddMeter("*")
+                .ConfigureResource(r => r.AddService("openfeature"))
+                .AddInMemoryExporter(exportedItems, option => option.PeriodicExportingMetricReaderOptions = new PeriodicExportingMetricReaderOptions { ExportIntervalMilliseconds = 100 })
```

Comment:
I'm glad you used the in-memory exporter for the tests. This is nicer than mocking everything IMO, since it means we don't have to make as many assumptions. I did the same thing in the JS impl.

Reply:
There was a small catch with that. Since we need to wait for the exporter, I needed to tune the periodic exporting to be more frequent and allow some buffer time for the metrics to be exported.

If this library was in net8, I would probably use the `IMeterFactory`: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.imeterfactory?view=net-8.0

Reply:
> There was a small catch with that. Since we need to wait for the exporter, I needed to tune the periodic exporting to be more frequent and allow some buffer time for the metrics to be exported.

I saw that. I think it's fine. The tests are still quick and I think the value in having an real exporter in the tests is worth a 150ms of waiting.